driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java

```diff
@@ -346,6 +371,64 @@ public boolean isClosed() {
     @Nullable
     @Override
     public <T> T sendAndReceive(final CommandMessage message, final Decoder<T> decoder, final OperationContext operationContext) {
+        try {
+            return sendAndReceiveInternal(message, decoder, operationContext);
+        } catch (MongoCommandException e) {
+            if (reauthenticationIsTriggered(e)) {
+                return reauthenticateAndRetry(()-> sendAndReceiveInternal(message, decoder, operationContext),
+                        operationContext);
+            }
+            throw e;
+        }
```

Comment:
Redundant code, and to match the async.

```suggestion
        Supplier<T> sendAndReceiveInternal = () -> sendAndReceiveInternal(
                message, decoder, operationContext);
        try {
            return sendAndReceiveInternal.get();
        } catch (MongoCommandException e) {
            if (reauthenticationIsTriggered(e)) {
                return reauthenticateAndRetry(sendAndReceiveInternal, operationContext);
            }
            throw e;
        }
```